crates/ty_python_semantic/resources/mdtest/dataclasses/dataclass_transform.md

```diff
@@ -195,7 +195,40 @@ OrderTrueOverwritten(1) < OrderTrueOverwritten(2)
 
 ### `kw_only_default`
 
-To do
+When provided, sets the default value for the `kw_only` parameter of `field()`.
+
+```py
+from typing import dataclass_transform
+from dataclasses import field
+
+@dataclass_transform(kw_only_default=True)
+def create_model(*, init=True): ...
+@create_model()
+class A:
+    name: str = field(default="Voldemort")
+
+a = A()
+a = A(name="Harry")
+a = A("Harry")  # error: [too-many-positional-arguments]
+```
+
+TODO: This can be overridden by the call to the decorator function.
+
+```py
+from typing import dataclass_transform
+
+@dataclass_transform(kw_only_default=True)
+def create_model(*, kw_only: bool = True): ...
+@create_model(kw_only=False)
+class CustomerModel:
+    id: int
+    name: str
+
+# TODO: Should not emit errors
```

Comment:
i believe this should cover the `dataclasses_transform_func` [false positive](https://github.com/python/typing/blob/d4f39b27a4a47aac8b6d4019e1b0b5b3156fabdc/conformance/tests/dataclasses_transform_func.py#L53)